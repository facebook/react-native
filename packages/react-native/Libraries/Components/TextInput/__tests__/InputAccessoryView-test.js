/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall react_native
 */

'use strict';

const render = require('../../../../jest/renderer');
const View = require('../../View/View');
const InputAccessoryView = require('../InputAccessoryView').default;
const React = require('react');

describe('InputAccessoryView', () => {
  it('should render as <RCTInputAccessoryView> when mocked', async () => {
    const instance = await render.create(
      <InputAccessoryView nativeID="1">
        <View />
      </InputAccessoryView>,
    );
    expect(instance).toMatchSnapshot();
  });

  it('should render as <RCTInputAccessoryView> when not mocked', async () => {
    jest.dontMock('../InputAccessoryView');

    const instance = await render.create(
      <InputAccessoryView nativeID="1">
        <View />
      </InputAccessoryView>,
    );
    expect(instance).toMatchSnapshot();
  });
});
