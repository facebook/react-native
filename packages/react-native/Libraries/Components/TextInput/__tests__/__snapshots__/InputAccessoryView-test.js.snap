// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAccessoryView should render as <RCTInputAccessoryView> when mocked 1`] = `
<RCTInputAccessoryView
  nativeID="1"
  style={
    Array [
      undefined,
      Object {
        "position": "absolute",
      },
    ]
  }
>
  <RCTSafeAreaView
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "width": 750,
        },
      ]
    }
  >
    <View />
  </RCTSafeAreaView>
</RCTInputAccessoryView>
`;

exports[`InputAccessoryView should render as <RCTInputAccessoryView> when not mocked 1`] = `
<RCTInputAccessoryView
  nativeID="1"
  style={
    Array [
      undefined,
      Object {
        "position": "absolute",
      },
    ]
  }
>
  <RCTSafeAreaView
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "width": 750,
        },
      ]
    }
  >
    <View />
  </RCTSafeAreaView>
</RCTInputAccessoryView>
`;
